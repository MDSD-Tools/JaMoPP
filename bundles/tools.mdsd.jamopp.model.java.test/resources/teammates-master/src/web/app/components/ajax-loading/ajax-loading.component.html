<div class="loading-container">
  <div class="spinner-border spinner-border-sm text-{{useBlueSpinner ? 'primary' : color}}" role="status"></div>
</div>
