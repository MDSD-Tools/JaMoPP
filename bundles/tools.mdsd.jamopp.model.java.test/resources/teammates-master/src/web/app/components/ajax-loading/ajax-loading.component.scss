.loading-container {
  display: inline;
  margin-right: 10px;
}
