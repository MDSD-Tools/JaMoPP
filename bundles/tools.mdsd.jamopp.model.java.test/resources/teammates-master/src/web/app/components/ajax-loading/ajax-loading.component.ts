import { Component, Input, OnInit } from '@angular/core';

/**
 * Displaying the ajax loader.
 */
@Component({
  selector: 'tm-ajax-loading',
  templateUrl: './ajax-loading.component.html',
  styleUrls: ['./ajax-loading.component.scss'],
})
export class AjaxLoadingComponent implements OnInit {

  @Input()
  useBlueSpinner: boolean = false;

  @Input()
  color: string = 'white';

  constructor() { }

  ngOnInit(): void {
  }

}
