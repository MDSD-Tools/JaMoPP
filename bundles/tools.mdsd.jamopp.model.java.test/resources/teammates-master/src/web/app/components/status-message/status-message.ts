/**
 * Represents a status message and a severity color.
 *
 * The color is adapted from Bootstrap's color scheme.
 */
export interface StatusMessage {
  message: string;
  color: string;
}
