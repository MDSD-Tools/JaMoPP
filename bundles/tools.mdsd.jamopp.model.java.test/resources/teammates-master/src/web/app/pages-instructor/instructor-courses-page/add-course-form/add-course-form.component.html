<div class="card border-primary fill-plain">
  <div class="close-header">
    <button type="button" class="close" aria-label="Close" (click)="closeEditFormHandler()">
      <span aria-hidden="true">&times;</span>
    </button>
  </div>
  <div class="row" >
    <div class="col-md-6">
      <div class="row align-items-center">
        <div class="col-md-6">
          <h5 class="text-center">Create new Course</h5>
        </div>
      </div>
    </div>
    <div class="col-md-6">
      <div class="row text-center align-items-center mt-1">
        <div class="col-md-3">
          <h5 class="text-md-right">Or: </h5>
        </div>
        <div class="col-md-9 text-md-left">
          <button id="btn-copy-course" type="button" class="btn btn-info" (click)="onCopy()" [disabled]="isCopyingCourse">
            <tm-ajax-loading *ngIf="isCopyingCourse"></tm-ajax-loading>
            Copy from previous courses
          </button>
        </div>
      </div>
    </div>
  </div>
  <div class="card-body text-center">
    <div class="form-group row">
      <label class="col-md-3 text-md-right col-form-label">Course ID:</label>
      <div class="col-md-6 text-md-left">
        <input id="new-course-id" class="form-control" type="text" [(ngModel)]="newCourseId"
            [maxlength]="64" [disabled]="isAddingCourse" placeholder="e.g. CS3215-2013Semester1"/>
      </div>
    </div>
    <div class="form-group row">
      <label class="col-md-3 text-md-right col-form-label">Course Name:</label>
      <div class="col-md-6 text-md-left">
        <input id="new-course-name" class="form-control" type="text" [(ngModel)]="newCourseName"
            [maxlength]="80" [disabled]="isAddingCourse" placeholder="e.g. Software Engineering"/>
      </div>
    </div>
    <div class="form-group row">
      <label class="col-md-3 text-md-right col-form-label ngb-tooltip-class"><span ngbTooltip="You should not
                need to change this as it is auto-detected based on your device settings. TEAMMATES automatically adjusts
                to match the current time offset in your area, including clock changes due to daylight saving time.">
        Time Zone:</span></label>
      <div class="col-md-6 text-md-left">
        <div class="input-group">
          <select id="new-time-zone" class="form-control" [(ngModel)]="timezone" [disabled]="isAddingCourse">
            <option *ngFor="let timezone of timezones" [value]="timezone.id">{{ timezone.id }} ({{ timezone.offset }})</option>
          </select>
          <span class="input-group-btn" style="margin-left: 5px;">
            <button class="btn btn-primary btn-margin-left" (click)="onAutoDetectTimezone()" [disabled]="isAddingCourse">Auto Detect</button>
          </span>
        </div>
      </div>
    </div>
    <div class="form-group row">
      <div class="col-md-9 offset-md-3 text-md-left">
        <button id="btn-save-course" class="btn btn-success" (click)="onSubmit()" [disabled]="isAddingCourse">
          <tm-ajax-loading *ngIf="isAddingCourse"></tm-ajax-loading>
          Add Course
        </button>
      </div>
    </div>
  </div>
</div>
