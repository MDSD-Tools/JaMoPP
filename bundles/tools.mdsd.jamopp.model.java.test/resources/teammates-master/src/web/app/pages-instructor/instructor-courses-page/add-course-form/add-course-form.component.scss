.fill-plain {
  background-color: #EAEFF5;
}

.close-header {
  padding-top: .75rem;
  padding-right: 1.25rem;
  margin-bottom: 0;
}

.col-form-label {
  font-weight: bold;
}
