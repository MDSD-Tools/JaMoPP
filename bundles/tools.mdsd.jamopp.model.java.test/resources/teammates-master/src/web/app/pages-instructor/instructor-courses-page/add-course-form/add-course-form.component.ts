import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
import { finalize } from 'rxjs/operators';
import { CourseService } from '../../../../services/course.service';
import { FeedbackSessionsService } from '../../../../services/feedback-sessions.service';
import { StatusMessageService } from '../../../../services/status-message.service';
import { TimezoneService } from '../../../../services/timezone.service';
import { Course, FeedbackSessions } from '../../../../types/api-output';
import { CopyCourseModalResult } from '../../../components/copy-course-modal/copy-course-modal-model';
import { CopyCourseModalComponent } from '../../../components/copy-course-modal/copy-course-modal.component';
import { ErrorMessageOutput } from '../../../error-message-output';

interface Timezone {
  id: string;
  offset: string;
}

const formatTwoDigits: Function = (n: number): string => {
  if (n < 10) {
    return `0${n}`;
  }
  return String(n);
};

/**
 * Instructor add new course form
 */
@Component({
  selector: 'tm-add-course-form',
  templateUrl: './add-course-form.component.html',
  styleUrls: ['./add-course-form.component.scss'],
})
export class AddCourseFormComponent implements OnInit {

  @Input() isEnabled: boolean = true;
  @Input() isCopyingCourse: boolean = false;
  @Input() activeCourses: Course[] = [];
  @Input() allCourses: Course[] = [];
  @Output() courseAdded: EventEmitter<void> = new EventEmitter<void>();
  @Output() closeCourseFormEvent: EventEmitter<void> = new EventEmitter<void>();
  @Output() copyCourseEvent: EventEmitter<CopyCourseModalResult> = new EventEmitter<CopyCourseModalResult>();

  timezones: Timezone[] = [];
  timezone: string = '';
  newCourseId: string = '';
  newCourseName: string = '';
  isAddingCourse: boolean = false;

  constructor(private statusMessageService: StatusMessageService,
              private courseService: CourseService,
              private timezoneService: TimezoneService,
              private feedbackSessionsService: FeedbackSessionsService,
              private ngbModal: NgbModal) { }

  ngOnInit(): void {
    for (const [id, offset] of Object.entries(this.timezoneService.getTzOffsets())) {
      const hourOffset: number = Math.floor(Math.abs(offset) / 60);
      const minOffset: number = Math.abs(offset) % 60;
      const sign: string = offset < 0 ? '-' : '+';
      this.timezones.push({
        id,
        offset: offset === 0 ? 'UTC' : `UTC ${sign}${formatTwoDigits(hourOffset)}:${formatTwoDigits(minOffset)}`,
      });
    }

    this.timezone = this.timezoneService.guessTimezone();
  }

  /**
   * Auto-detects timezone for instructor.
   */
  onAutoDetectTimezone(): void {
    if (!this.isEnabled) {
      return;
    }
    this.timezone = this.timezoneService.guessTimezone();
  }

  /**
   * Submits the data to add the new course.
   */
  onSubmit(): void {
    if (!this.isEnabled) {
      return;
    }
    if (!this.newCourseId || !this.newCourseName) {
      this.statusMessageService.showErrorToast(
          'Please make sure you have filled in both Course ID and Name before adding the course!');
      return;
    }

    this.isAddingCourse = true;
    this.courseService.createCourse({
      courseName: this.newCourseName,
      timeZone: this.timezone,
      courseId: this.newCourseId,
    }).pipe(finalize(() => this.isAddingCourse = false)).subscribe(() => {
      this.courseAdded.emit();
      this.statusMessageService.showSuccessToast('The course has been added.');
    }, (resp: ErrorMessageOutput) => {
      this.statusMessageService.showErrorToast(resp.error.message);
    });
    this.newCourseId = '';
    this.newCourseName = '';
    this.timezone = this.timezoneService.guessTimezone();
  }

  /**
   * Handles closing of the edit form.
   */
  closeEditFormHandler(): void {
    this.closeCourseFormEvent.emit();
  }

  /**
   * Handles copying from other courses event
   */
  onCopy(): void {
    const modalRef: NgbModalRef = this.ngbModal.open(CopyCourseModalComponent);
    modalRef.componentInstance.isCopyFromOtherSession = true;
    modalRef.componentInstance.allCourses = this.allCourses;
    modalRef.componentInstance.activeCourses = this.activeCourses;

    modalRef.componentInstance.fetchFeedbackSessionsEvent.subscribe((courseId: string) => {
      this.feedbackSessionsService
        .getFeedbackSessionsForInstructor(courseId)
        .subscribe((feedbackSessions: FeedbackSessions) => {
          modalRef.componentInstance.courseToFeedbackSession[courseId] = [...feedbackSessions.feedbackSessions];
        });
    }, (resp: ErrorMessageOutput) => {
      this.statusMessageService.showErrorToast(resp.error.message);
    });

    modalRef.result.then((result: CopyCourseModalResult) => this.copyCourseEvent.emit(result),
      (resp: ErrorMessageOutput) => this.statusMessageService.showErrorToast(resp.error.message));
  }
}
