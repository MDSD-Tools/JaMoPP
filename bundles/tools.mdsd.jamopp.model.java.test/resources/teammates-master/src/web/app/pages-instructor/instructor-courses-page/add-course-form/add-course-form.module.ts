import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NgbTooltipModule } from '@ng-bootstrap/ng-bootstrap';
import { AjaxLoadingModule } from '../../../components/ajax-loading/ajax-loading.module';
import { AddCourseFormComponent } from './add-course-form.component';

/**
 * Module for form for adding courses.
 */
@NgModule({
  declarations: [AddCourseFormComponent],
  exports: [AddCourseFormComponent],
  imports: [
    CommonModule,
    FormsModule,
    RouterModule,
    AjaxLoadingModule,
    NgbTooltipModule,
  ],
})
export class AddCourseFormModule { }
