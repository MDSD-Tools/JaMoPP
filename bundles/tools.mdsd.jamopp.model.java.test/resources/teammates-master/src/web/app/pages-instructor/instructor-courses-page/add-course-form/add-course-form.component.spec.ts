import { HttpClientTestingModule } from '@angular/common/http/testing';
import { EventEmitter } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';
import { NgbModal, NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { of } from 'rxjs';

import { CopyCourseModalComponent } from '../../../../app/components/copy-course-modal/copy-course-modal.component';
import { CourseService } from '../../../../services/course.service';
import { FeedbackSessionsService } from '../../../../services/feedback-sessions.service';
import { StatusMessageService } from '../../../../services/status-message.service';
import { TimezoneService } from '../../../../services/timezone.service';
import {
  Course,
  FeedbackSession,
  FeedbackSessionPublishStatus,
  FeedbackSessionSubmissionStatus,
  ResponseVisibleSetting,
  SessionVisibleSetting,
} from '../../../../types/api-output';
import { AjaxLoadingModule } from '../../../components/ajax-loading/ajax-loading.module';
import { AddCourseFormComponent } from './add-course-form.component';

describe('AddCourseFormComponent', () => {
  let component: AddCourseFormComponent;
  let fixture: ComponentFixture<AddCourseFormComponent>;
  let feedbackSessionsService: FeedbackSessionsService;
  let ngbModal: NgbModal;

  const testCourseId: string = 'CS3281';
  const testCourseName: string = 'Valid course';
  const testTimeZone: string = 'UTC';
  const timeZoneOffsets1: Record<string, number> = { GMT: 0 };
  const testCourse: Course = {
    courseId: 'testId',
    courseName: 'Test Course',
    timeZone: 'Asia/Singapore',
    institute: 'Test Institute',
    creationTimestamp: 0,
    deletionTimestamp: 1000,
  };

  const spyStatusMessageService: any = {
    showErrorToast: jest.fn(),
    showSuccessToast: jest.fn(),
  };
  const timezoneServiceStub: any = {
    getTzOffsets: jest.fn(() => timeZoneOffsets1),
    guessTimezone: jest.fn(() => 'UTC'),
  };
  const spyCourseService: any = {
    createCourse: jest.fn(() => of({})),
    getAllCoursesAsInstructor: jest.fn(() => of({ courses: [testCourse] })),
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AddCourseFormComponent],
      imports: [
        HttpClientTestingModule,
        FormsModule,
        ReactiveFormsModule,
        RouterTestingModule,
        NgbModule,
        AjaxLoadingModule,
      ],
      providers: [
        { provide: StatusMessageService, useValue: spyStatusMessageService },
        { provide: CourseService, useValue: spyCourseService },
        { provide: TimezoneService, useValue: timezoneServiceStub },
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AddCourseFormComponent);
    feedbackSessionsService = TestBed.inject(FeedbackSessionsService);
    ngbModal = TestBed.inject(NgbModal);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    // Unable to leave timezone as default field, otherwise the field defaults to the
    // timezone the system is on. This will differ from
    // place to place causing the snapshot to constantly be mismatched.
    component.timezone = testTimeZone;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when not enabled', () => {
    component.isEnabled = false;
    component.timezone = testTimeZone;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should call showErrorToast when courseId is blank', () => {
    component.newCourseId = '';
    component.onSubmit();
    fixture.detectChanges();
    expect(spyStatusMessageService.showErrorToast).toHaveBeenCalled();
  });

  it('should hold added course with valid details', () => {
    component.newCourseId = testCourseId;
    component.newCourseName = testCourseName;
    component.onSubmit();
    fixture.detectChanges();
    expect(spyStatusMessageService.showSuccessToast).toHaveBeenCalled();
  });

  it('should open copy course modal', () => {
    class MockNgbModalRef {
      componentInstance: any = {
        isCopyFromOtherSession: false,
        courses: [],
        courseToFeedbackSession: {},
        fetchFeedbackSessionsEvent: new EventEmitter<string>(),
      };
      result: Promise<any> = Promise.resolve();
    }

    const mockModalRef: MockNgbModalRef = new MockNgbModalRef();

    const testFeedbackSession: FeedbackSession = {
      courseId: 'testId',
      timeZone: 'Asia/Singapore',
      feedbackSessionName: 'Test Session',
      instructions: 'Instructions',
      submissionStartTimestamp: 1000000000000,
      submissionEndTimestamp: 1500000000000,
      gracePeriod: 0,
      sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
      responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
      submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
      publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
      isClosingEmailEnabled: true,
      isPublishedEmailEnabled: true,
      createdAtTimestamp: 0,
    };

    spyOn(feedbackSessionsService, 'getFeedbackSessionsForInstructor')
      .and.returnValue(of({ feedbackSessions: [testFeedbackSession] }));
    spyOn(ngbModal, 'open').and.returnValue(mockModalRef);
    component.activeCourses = [testCourse];

    component.onCopy();
    mockModalRef.componentInstance.fetchFeedbackSessionsEvent.emit(testCourse.courseId);

    expect(ngbModal.open).toHaveBeenCalledWith(CopyCourseModalComponent);
    expect(mockModalRef.componentInstance.isCopyFromOtherSession).toEqual(true);
    expect(mockModalRef.componentInstance.activeCourses[0]).toEqual(testCourse);
    expect(mockModalRef.componentInstance.courseToFeedbackSession[testCourse.courseId]).toEqual([testFeedbackSession]);
  });
});
