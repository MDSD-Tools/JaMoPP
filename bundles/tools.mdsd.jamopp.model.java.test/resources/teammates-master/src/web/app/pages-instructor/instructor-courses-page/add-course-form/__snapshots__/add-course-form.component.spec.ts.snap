// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseFormComponent should snap when not enabled 1`] = `
<tm-add-course-form
  activeCourses={[Function Array]}
  allCourses={[Function Array]}
  closeCourseFormEvent={[Function EventEmitter]}
  copyCourseEvent={[Function EventEmitter]}
  courseAdded={[Function EventEmitter]}
  courseService={[Function Object]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  isAddingCourse="false"
  isCopyingCourse="false"
  isEnabled="false"
  newCourseId=""
  newCourseName=""
  ngbModal={[Function NgbModal]}
  statusMessageService={[Function Object]}
  timezone={[Function String]}
  timezoneService={[Function Object]}
  timezones={[Function Array]}
>
  <div
    class="card border-primary fill-plain"
  >
    <div
      class="close-header"
    >
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col-md-6"
          >
            <h5
              class="text-center"
            >
              Create new Course
            </h5>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="row text-center align-items-center mt-1"
        >
          <div
            class="col-md-3"
          >
            <h5
              class="text-md-right"
            >
              Or: 
            </h5>
          </div>
          <div
            class="col-md-9 text-md-left"
          >
            <button
              class="btn btn-info"
              id="btn-copy-course"
              type="button"
            >
              
               Copy from previous courses 
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card-body text-center"
    >
      <div
        class="form-group row"
      >
        <label
          class="col-md-3 text-md-right col-form-label"
        >
          Course ID:
        </label>
        <div
          class="col-md-6 text-md-left"
        >
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="new-course-id"
            maxlength="64"
            placeholder="e.g. CS3215-2013Semester1"
            type="text"
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="col-md-3 text-md-right col-form-label"
        >
          Course Name:
        </label>
        <div
          class="col-md-6 text-md-left"
        >
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="new-course-name"
            maxlength="80"
            placeholder="e.g. Software Engineering"
            type="text"
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="col-md-3 text-md-right col-form-label ngb-tooltip-class"
        >
          <span
            ngbtooltip="You should not
                  need to change this as it is auto-detected based on your device settings. TEAMMATES automatically adjusts
                  to match the current time offset in your area, including clock changes due to daylight saving time."
          >
             Time Zone:
          </span>
        </label>
        <div
          class="col-md-6 text-md-left"
        >
          <div
            class="input-group"
          >
            <select
              class="form-control ng-untouched ng-pristine ng-valid"
              id="new-time-zone"
            >
              
              <option
                value="GMT"
              >
                GMT (UTC)
              </option>
            </select>
            <span
              class="input-group-btn"
              style="margin-left: 5px;"
            >
              <button
                class="btn btn-primary btn-margin-left"
              >
                Auto Detect
              </button>
            </span>
          </div>
        </div>
      </div>
      <div
        class="form-group row"
      >
        <div
          class="col-md-9 offset-md-3 text-md-left"
        >
          <button
            class="btn btn-success"
            id="btn-save-course"
          >
            
             Add Course 
          </button>
        </div>
      </div>
    </div>
  </div>
</tm-add-course-form>
`;

exports[`AddCourseFormComponent should snap with default fields 1`] = `
<tm-add-course-form
  activeCourses={[Function Array]}
  allCourses={[Function Array]}
  closeCourseFormEvent={[Function EventEmitter]}
  copyCourseEvent={[Function EventEmitter]}
  courseAdded={[Function EventEmitter]}
  courseService={[Function Object]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  isAddingCourse="false"
  isCopyingCourse="false"
  isEnabled={[Function Boolean]}
  newCourseId=""
  newCourseName=""
  ngbModal={[Function NgbModal]}
  statusMessageService={[Function Object]}
  timezone={[Function String]}
  timezoneService={[Function Object]}
  timezones={[Function Array]}
>
  <div
    class="card border-primary fill-plain"
  >
    <div
      class="close-header"
    >
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col-md-6"
          >
            <h5
              class="text-center"
            >
              Create new Course
            </h5>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="row text-center align-items-center mt-1"
        >
          <div
            class="col-md-3"
          >
            <h5
              class="text-md-right"
            >
              Or: 
            </h5>
          </div>
          <div
            class="col-md-9 text-md-left"
          >
            <button
              class="btn btn-info"
              id="btn-copy-course"
              type="button"
            >
              
               Copy from previous courses 
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card-body text-center"
    >
      <div
        class="form-group row"
      >
        <label
          class="col-md-3 text-md-right col-form-label"
        >
          Course ID:
        </label>
        <div
          class="col-md-6 text-md-left"
        >
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="new-course-id"
            maxlength="64"
            placeholder="e.g. CS3215-2013Semester1"
            type="text"
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="col-md-3 text-md-right col-form-label"
        >
          Course Name:
        </label>
        <div
          class="col-md-6 text-md-left"
        >
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="new-course-name"
            maxlength="80"
            placeholder="e.g. Software Engineering"
            type="text"
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="col-md-3 text-md-right col-form-label ngb-tooltip-class"
        >
          <span
            ngbtooltip="You should not
                  need to change this as it is auto-detected based on your device settings. TEAMMATES automatically adjusts
                  to match the current time offset in your area, including clock changes due to daylight saving time."
          >
             Time Zone:
          </span>
        </label>
        <div
          class="col-md-6 text-md-left"
        >
          <div
            class="input-group"
          >
            <select
              class="form-control ng-untouched ng-pristine ng-valid"
              id="new-time-zone"
            >
              
              <option
                value="GMT"
              >
                GMT (UTC)
              </option>
            </select>
            <span
              class="input-group-btn"
              style="margin-left: 5px;"
            >
              <button
                class="btn btn-primary btn-margin-left"
              >
                Auto Detect
              </button>
            </span>
          </div>
        </div>
      </div>
      <div
        class="form-group row"
      >
        <div
          class="col-md-9 offset-md-3 text-md-left"
        >
          <button
            class="btn btn-success"
            id="btn-save-course"
          >
            
             Add Course 
          </button>
        </div>
      </div>
    </div>
  </div>
</tm-add-course-form>
`;
